ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/BatchedValidatingCommitter.scala

```diff
@@ -18,6 +18,37 @@ import com.daml.ledger.validator.batch.{
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
+/**
+  * Orchestrates committing to a ledger after validating submissions. Supports parallel validation.
+  * Example usage, assuming a [[com.daml.ledger.participant.state.kvutils.api.BatchingLedgerWriter]] sends the
+  * batched submissions over the wire:
+  * {{{
```

Comment:
Minor comment: I generally try to avoid code blocks in scaladoc comments because they quickly tend to become outdated, as they are not tested in CI. I'm sure there's an example implementation somewhere, e.g., in `com.daml.ledger.on.memory`. Would it make sense to move this example there and just set a pointer here?

Reply:
Unfortunately, there's no good example for the general use-case hence I added it here. Such examples going out of date is a good point -- I'll look into how these could be validated in an automated way.